scripts/init.sql

```diff
@@ -1,187 +1,176 @@
-        "type" integer NOT NULL,
-        expiration timestamp NOT NULL,
-        user_id integer NOT NULL,
-        created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
-        CONSTRAINT verification_tokens_pkey PRIMARY KEY(id)
-    );
-
-CREATE TABLE
-    IF NOT EXISTS verifications(
-        id integer NOT NULL,
-        email_verified boolean NOT NULL,
-        user_id integer NOT NULL,
-        created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
-        updated_at timestamp,
-        CONSTRAINT verifications_pkey PRIMARY KEY(id)
-    );
-
-CREATE TABLE
-    IF NOT EXISTS wishcards(
-        id integer NOT NULL,
-        address_line_1 varchar(255) NOT NULL,
-        address_line_2 varchar(255),
-        city varchar(100) NOT NULL,
-        state varchar(100) NOT NULL,
-        country varchar(100) NOT NULL,
-        zip_code varchar(50) NOT NULL,
-        status integer NOT NULL DEFAULT 0,
-        created_by integer NOT NULL,
-        agency_id integer NOT NULL,
-        child_id integer NOT NULL,
-        item_id integer NOT NULL,
-        image_id integer NOT NULL,
-        order_id integer NOT NULL,
-        created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
-        updated_at timestamp,
-        CONSTRAINT wishcards_pkey PRIMARY KEY(id)
-    );
-
--- constraints
+SET check_function_bodies = false ;
+
+CREATE TABLE agencies(
+    id uuid NOT NULL DEFAULT gen_random_uuid(),
+    "name" varchar(255) NOT NULL,
+    website varchar(255) NOT NULL,
+    phone varchar(100) NOT NULL,
+    bio varchar(500),
+    address_line_1 varchar(255) NOT NULL,
+    address_line_2 varchar(255),
+    city varchar(100) NOT NULL,
+    state varchar(100) NOT NULL,
+    country varchar(100) NOT NULL,
+    zip_code varchar(50) NOT NULL,
+    verified boolean NOT NULL DEFAULT false,
+    employer_identification_number varchar(255),
+    account_manager_id uuid NOT NULL,
+    image_id uuid NOT NULL,
+    created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updated_at timestamp,
+    CONSTRAINT agencies_pkey PRIMARY KEY(id),
+    UNIQUE(employer_identification_number)
+);
+
+CREATE TABLE children(
+    id uuid NOT NULL DEFAULT gen_random_uuid(),
+    first_name varchar(255) NOT NULL,
+    last_name varchar(255) NOT NULL,
+    birth_year integer NOT NULL,
+    interest varchar(255) NOT NULL,
+    story varchar(255) NOT NULL,
+    image_id uuid NOT NULL,
+    created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    CONSTRAINT children_pkey PRIMARY KEY(id)
+);
+
+CREATE TABLE community_posts(
+    id uuid NOT NULL DEFAULT gen_random_uuid(),
+    message varchar(255) NOT NULL,
+    agency_id uuid NOT NULL,
+    image_id uuid NOT NULL,
+    created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    CONSTRAINT community_posts_pkey PRIMARY KEY(id)
+);
+
+CREATE TABLE images(
+    id uuid NOT NULL DEFAULT gen_random_uuid(),
+    url varchar(255) NOT NULL,
+    meta_data json,
+    created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
```

Comment:
```suggestion
    created_by uuid NOT NULL,
    created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
```

@Enubia Thoughts on adding a `created_by` field on the `images` table? We do technically need audit logging for a majority of our other tables anyways, but since images take up more resources than most things, it may be helpful if we can determine which user uploaded any given image in case we run into a situation where a bad actor tries to DOS us by spam uploading images. That way, we can apply a fix to prevent the attack and can also selectively delete all images uploaded by the individual.

We technically have a reference to the current image for the user/wishcard on those tables, but if they upload multiple images it breaks the 1 to 1 connection between them.

Reply:
If we need it in more places wouldn't a revisions table make more sense?
This way we can set up a trigger I assume to log every interaction that has taken place by a user

Would also be useful for the back office so that we know who did what at any given time

Reply:
Yes, an audit table is usually very beneficial to have for a multitude of things, especially trailing during debugging as well as for notifications.

I don't think we should implement one yet, but it definitely would be an easy win post-kysely integration and database migration.

Reply:
I think @nick-w-nick has a good point!

Reply:
I've added the created_by field with a one-to-many relation since a user can add images for an agency, wish card, child and himself